compiler/optimizer/RegisterCandidate.cpp

```diff
@@ -608,11 +608,12 @@ TR_RegisterCandidate::setWeight(TR::Block * * blocks, int32_t *blockStructureWei
        //
       if (extendedBlockFreq <= blockFreq)
          {
-         if ((uint32_t)blockWeight > (*_loadsAndStores)[extendedBlock->getNumber()])
-            (*_loadsAndStores)[extendedBlock->getNumber()] = blockWeight;
+         if ((uint32_t)blockWeight > _blocks.getNumberOfLoadsAndStores(extendedBlock->getNumber()))//(*_loadsAndStores)[extendedBlock->getNumber()])
+            //(*_loadsAndStores)[extendedBlock->getNumber()] = blockWeight;
+            _blocks.setNumberOfLoadsAndStores(extendedBlock->getNumber(), blockWeight);
          }
-      if ((uint32_t)blockWeight > (*_loadsAndStores)[blockNumber])
-         (*_loadsAndStores)[blockNumber] = blockWeight;
+      //if ((uint32_t)blockWeight > (*_loadsAndStores)[blockNumber])
+      //   (*_loadsAndStores)[blockNumber] = blockWeight;
```

Comment:
The change is is more subtle that this if disappearing. the _loadAndStores structure was, to a large extent, duplicating the information in the _blocks data structure. This if was copying the data from _blocks to _loadAndStores. The difference here is that there is an accumulation of frequencies over part of the extended basic block for _loadAndStores that does not happen for _blocks before my change. The change to accumulate the info in _blocks seemed to be logical and unlikely to affect current usage while saving the duplicate data structure. Performance testing so far say there is no impact of throughput performance. The semantic change is due to the unification of _blocks and _loadAndStores.